task2-3/src/test/unit/routers/groupRouter.spec.ts

```diff
@@ -0,0 +1,146 @@
+import request from 'supertest';
+import { NextFunction, Request, Response } from 'express';
+import groupService from '../../../services/groupService';
+import { createServer } from '../../createServer';
+import { group, groups } from '../../mocks/groupsMock';
+import { groupRouter } from '../../../routers/groupRouter';
+
+jest.mock('../../../services/groupService', () => ({
+    editGroup: jest.fn(),
+    getGroupById: jest.fn(),
+    postGroup: jest.fn(),
+    deleteGroup: jest.fn(),
+    getAllGroups: jest.fn(),
+    postUsersToGroup: jest.fn()
+}));
+
+jest.mock('../../../middleware/validateToken', () => ({
+    validateToken: (req: Request, res: Response, next: NextFunction) => {
+        return next();
+    }
+}));
+
+jest.mock('uuid', () => ({
+    v4: () => '1'
+}));
+
+beforeEach(() => {
+    jest.spyOn(groupService, 'editGroup').mockResolvedValue({} as any);
+    jest.spyOn(groupService, 'deleteGroup').mockResolvedValue({} as any);
+    jest.spyOn(groupService, 'getGroupById').mockResolvedValue(group as any);
+    jest.spyOn(groupService, 'getAllGroups').mockResolvedValue(groups);
+    jest.spyOn(groupService, 'postGroup').mockResolvedValue({} as any);
+    jest.spyOn(groupService, 'postUsersToGroup').mockResolvedValue({} as any);
+});
+
+afterEach(() => {
+    jest.resetAllMocks();
+});
+
+describe('Group router tests', () => {
+    const app = createServer((application: any) => {
+        application.use('/groups', groupRouter);
+    });
+
+    it('should POST group', async () => {
```

Comment:
It is not the best test name :) I don't understand what the test result you want to verify nere

Reply:
applicable to all tests in the file